services/self-service/src/test/java/com/epam/dlab/backendapi/resources/ProjectResourceTest.java

```diff
@@ -50,81 +58,126 @@
-
-		assertEquals(HttpStatus.SC_ACCEPTED, response.getStatus());
-		verify(projectService).start(any(UserInfo.class), anyList(), anyString());
-		verifyNoMoreInteractions(projectService);
-	}
-
-	@Test
-	public void generate() {
-		when(keyService.generateKeys(any(UserInfo.class))).thenReturn(new KeysDTO("somePublicKey", "somePrivateKey",
-				"user"));
-
-		final Response response = resources.getJerseyTest()
-				.target("/project/keys")
-				.request()
-				.header("Authorization", "Bearer " + TOKEN)
-				.post(Entity.json(""));
-
-		assertEquals(HttpStatus.SC_OK, response.getStatus());
-		assertEquals(MediaType.APPLICATION_JSON, response.getHeaderString(HttpHeaders.CONTENT_TYPE));
-
-		verify(keyService).generateKeys(getUserInfo());
-		verifyNoMoreInteractions(keyService);
-	}
-
-	@Test
-	public void generateKeysWithException() {
-		doThrow(new DlabException("Can not generate private/public key pair due to"))
-				.when(keyService).generateKeys(any(UserInfo.class));
-
-		final Response response = resources.getJerseyTest()
-				.target("/project/keys")
-				.request()
-				.header("Authorization", "Bearer " + TOKEN)
-				.post(Entity.json(""));
-
-		assertEquals(HttpStatus.SC_INTERNAL_SERVER_ERROR, response.getStatus());
-		assertEquals(MediaType.APPLICATION_JSON, response.getHeaderString(HttpHeaders.CONTENT_TYPE));
-
-		verify(keyService).generateKeys(getUserInfo());
-		verifyNoMoreInteractions(keyService);
-	}
-
-	private ProjectActionFormDTO getProjectActionDTO() {
-		return new ProjectActionFormDTO("DLAB", Collections.singletonList("https://localhost:8083/"));
-	}
+    private ProjectService projectService = mock(ProjectService.class);
+    private AccessKeyService keyService = mock(AccessKeyService.class);
+    private String PROJECT_NAME = "DLAB";
+
+    @Rule
+    public final ResourceTestRule resources = getResourceTestRuleInstance(
+            new ProjectResource(projectService, keyService));
+
+    @Before
+    public void setup() throws AuthenticationException {
+        authSetup();
+    }
+
+    @Test
+    public void createProject() {
+        CreateProjectDTO createProjectDTO = returnCreateProjectDTO();
+        final Response response = resources.getJerseyTest()
+                .target("project")
+                .request()
+                .header("Authorization", "Bearer " + TOKEN)
+                .post(Entity.json(createProjectDTO));
+
+        assertEquals(HttpStatus.SC_OK, response.getStatus());
+        verify(projectService).create(getUserInfo(), returnProjectDTO(createProjectDTO), createProjectDTO.getName());
+        verifyNoMoreInteractions(projectService);
+    }
+
+    @Test
+    public void createExistingProject() {
+        CreateProjectDTO createProjectDTO = returnCreateProjectDTO();
+        doThrow(new ResourceConflictException("Project with passed name already exist in system"))
+                .when(projectService).create(getUserInfo(), returnProjectDTO(createProjectDTO), createProjectDTO.getName());
```

Comment:
Here you should specify **any()** methods(any(), anyString(), any(Class.class))

Reply:
methods has been changed